modules/kernel/src/org/apache/axis2/deployment/DescriptionBuilder.java

```diff
@@ -691,15 +698,43 @@ protected String getValue(String in) {
     }
 
     protected String replaceSystemProperty(String text) {
+        String[] sysRefs = StringUtils.substringsBetween(text, SYS_PROPERTY_PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+        String[] envRefs = StringUtils.substringsBetween(text, ENV_VAR_PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+
+        // Resolves system property references ($sys{ref}) in an individual string.
+        if (sysRefs != null) {
+            for (String ref : sysRefs) {
```

Comment:
Ideally, there can be only 1 ref, right? Shall we add a validation for the size? just to make sure we will not have any property of the form $sys{$sys{something}} for some kind of a bug. The same goes with environment variables

Reply:
Adding a size validation will restrict us from defining a property of the form $sys{prop1}$sys{prop2}. If a property is specified in the form $sys{$sys{something}}, we will end up having only a single element (invalid) in the array.

Reply:
Do we need anything of the form $sys{prop1}$sys{prop2}? Seems a bit odd to me. @Nirothipan @nilminiwso2 WDYT?

Reply:
This seems okay for since the user can provide single value via chaining multiple properties.

This should work for property like ${user}-${user-home} , @Vathsan ?

But this behavior should be consistent, in other places also we should support chaining of properties.

Reply:
Then we need to add it to the existing ${key} functionality aswell. I feel like this is a bit too complicating stuff. If the user needs to chain properties, he can always do it and assign it to a single variable.

Reply:
Updated the method to resolve only one ref in 7bc767446d2e485f2bfd7f274684d8137f323ef4, as we are not going to support chaining multiple properties.